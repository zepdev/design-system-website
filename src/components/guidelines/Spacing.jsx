import React from 'react'
import clsx from 'clsx'
import { createUseStyles, useTheme } from 'react-jss'
import { spacing } from 'zeppelin-element-library/guidelines.json'

let useStyles = createUseStyles(theme => ({
  root: {
    overflowX: 'auto',
  },
  table: {
    width: '100%',
    marginBottom: theme.spacing.component.xl.px * 2,
    borderCollapse: 'collapse',
  },
  th: {
    padding: `${theme.spacing.component.m.px}px ${theme.spacing.component.l.px}px`,
    textAlign: 'left',
    display: 'table-cell',
  },
  tr: {
    display: 'table-row',
  },
  tbody: {
    display: 'table-row-group',
  },
  header: {
    background: theme.color.gray.grayLightest.hex,
    display: 'table-header-group',
  },
  lowercase: {
    textTransform: 'lowercase',
  },
  capitalize: {
    textTransform: 'capitalize',
  },
  borderBottom: {
    borderBottom: `1px solid ${theme.color.gray.grayLighter.hex}`,
  },
  square: {
    background: theme.theme.indigo.primary,
  },
}))

const Spacing = ({ ...props }) => {
  const theme = useTheme()
  const classes = useStyles({ ...props, theme })
  const header = ['name', 'px', 'rem', 'actual size']

  return (
    <div className={classes.root}>
      <table className={classes.table}>
        <caption className="zep-visually-hidden">Spacing Scale</caption>
        <thead className={classes.header}>
          <tr className={classes.tr}>
            {header.map(elem => (
              <th
                scope="col"
                className={clsx(classes.th, 'zep-typo--normal-1', {
                  [classes.lowercase]: elem === 'rem' || elem === 'px',
                  [classes.capitalize]: elem !== 'rem' && elem !== 'px',
                })}
                key={`header_${elem}`}
              >
                {elem}
              </th>
            ))}
          </tr>
        </thead>
        <tbody className={classes.tbody}>
          {Object.keys(spacing.component).map(elem => (
            <tr key={`row_${elem}`} className={classes.tr}>
              {header.map((item, idx) => (
                <td
                  className={clsx(
                    classes.th,
                    classes.borderBottom,
                    'zep-typo--normal-body2'
                  )}
                  key={`spacing${idx}`}
                  scope={idx === 0 ? 'row' : null}
                >
                  {item === 'actual size' ? (
                    <div
                      style={{
                        width: spacing.component[elem].px,
                        height: spacing.component[elem].px,
                        color: 'green',
                      }}
                      className={classes.square}
                    />
                  ) : item === 'name' ? (
                    `zep-spacing--${elem}`
                  ) : item === 'px' ? (
                    `${spacing.component[elem][item]}px`
                  ) : (
                    `${spacing.component[elem][item]}rem`
                  )}
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  )
}

export default Spacing
