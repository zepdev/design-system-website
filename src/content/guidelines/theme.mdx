---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
