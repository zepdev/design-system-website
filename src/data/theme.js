module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
