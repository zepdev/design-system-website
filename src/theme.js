import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
