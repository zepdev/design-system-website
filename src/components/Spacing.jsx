import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import withStyles from 'react-jss'
import theme from '../data/theme'

const styles = theme => ({
  table: {
    width: '100%',
    marginBottom: theme.spacing.xl.px * 2,
    borderCollapse: 'collapse',
    overflowX: 'scroll',
  },
  th: {
    padding: `${ theme.spacing.m.px } ${ theme.spacing.l.px }`,
    textAlign: 'left',
    display: 'table-cell',
  },
  tr: {
    display: 'table-row',
  },
  tbody: {
    display: 'table-row-group',
  },
  header: {
    background: theme.colors.gray.grayLightest.hex,
    display: 'table-header-group',
  },
  lowercase: {
    textTransform: 'lowercase',
  },
  capitalize: {
    textTransform: 'capitalize',
  },
  borderBottom: {
    borderBottom: `1px solid ${ theme.colors.gray.grayLighter.hex }`,
  },
  square: {
    background: theme.colors.primary.indigoBlue.hex,
  },
})

const Spacing = ({ classes }) => {
  const header = ['name', 'px', 'rem', 'actual size']

  return (
    <table className={classes.table}>
      <caption className="visually-hidden">Spacing Scale</caption>
      <thead className={classes.header}>
        <tr className={classes.tr}>
          {header.map(elem => (
            <th
              scope="col"
              className={classnames(classes.th, 'zep-typo--normal-8', {
                [classes.lowercase]: elem === 'rem' || elem === 'px',
                [classes.capitalize]: elem !== 'rem' && elem !== 'px',
              })}
              key={`header_${ elem }`}
            >
              {elem}
            </th>
          ))}
        </tr>
      </thead>
      <tbody className={classes.tbody}>
        {Object.keys(theme.spacing).map(elem => (
          <tr key={`row_${ elem }`} className={classes.tr}>
            {header.map((item, idx) => (
              <td
                className={classnames(classes.th, classes.borderBottom, 'zep-typo--normal-7')}
                key={`spacing${ idx }`}
                scope={idx === 0 ? 'row' : null}
              >
                {item === 'actual size' ? (
                  <div
                    style={{
                      width: theme.spacing[elem].px,
                      height: theme.spacing[elem].px,
                      color: 'green',
                    }}
                    className={classes.square}
                  />
                ) : item === 'name' ? (
                  `zep-spacing--${ elem }`
                ) : (
                  theme.spacing[elem][item]
                )}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  )
}

Spacing.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Spacing)
