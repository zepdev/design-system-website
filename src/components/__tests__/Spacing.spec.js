import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'react-jss'
import theme from '../../data/theme'
import Spacing from '../Spacing'

describe('Spacing', () => {
  it('renders correctly', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Spacing />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
