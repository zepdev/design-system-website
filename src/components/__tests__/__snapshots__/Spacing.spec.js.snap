// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacing renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="Spacing-table-0-1-1"
      >
        <caption
          class="visually-hidden"
        >
          Spacing Scale
        </caption>
        <thead
          class="Spacing-header-0-1-5"
        >
          <tr
            class="Spacing-tr-0-1-3"
          >
            <th
              class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-capitalize-0-1-7"
              scope="col"
            >
              name
            </th>
            <th
              class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-lowercase-0-1-6"
              scope="col"
            >
              px
            </th>
            <th
              class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-lowercase-0-1-6"
              scope="col"
            >
              rem
            </th>
            <th
              class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-capitalize-0-1-7"
              scope="col"
            >
              actual size
            </th>
          </tr>
        </thead>
        <tbody
          class="Spacing-tbody-0-1-4"
        >
          <tr
            class="Spacing-tr-0-1-3"
          >
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
              scope="row"
            >
              zep-spacing--xxs
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              4px
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              0.25rem
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              <div
                class="Spacing-square-0-1-9"
                style="width: 4px; height: 4px; color: green;"
              />
            </td>
          </tr>
          <tr
            class="Spacing-tr-0-1-3"
          >
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
              scope="row"
            >
              zep-spacing--xs
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              8px
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              0.5rem
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              <div
                class="Spacing-square-0-1-9"
                style="width: 8px; height: 8px; color: green;"
              />
            </td>
          </tr>
          <tr
            class="Spacing-tr-0-1-3"
          >
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
              scope="row"
            >
              zep-spacing--s
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              12px
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              0.75rem
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              <div
                class="Spacing-square-0-1-9"
                style="width: 12px; height: 12px; color: green;"
              />
            </td>
          </tr>
          <tr
            class="Spacing-tr-0-1-3"
          >
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
              scope="row"
            >
              zep-spacing--m
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              16px
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              1rem
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              <div
                class="Spacing-square-0-1-9"
                style="width: 16px; height: 16px; color: green;"
              />
            </td>
          </tr>
          <tr
            class="Spacing-tr-0-1-3"
          >
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
              scope="row"
            >
              zep-spacing--l
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              24px
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              1.5rem
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              <div
                class="Spacing-square-0-1-9"
                style="width: 24px; height: 24px; color: green;"
              />
            </td>
          </tr>
          <tr
            class="Spacing-tr-0-1-3"
          >
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
              scope="row"
            >
              zep-spacing--xl
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              32px
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              2rem
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              <div
                class="Spacing-square-0-1-9"
                style="width: 32px; height: 32px; color: green;"
              />
            </td>
          </tr>
          <tr
            class="Spacing-tr-0-1-3"
          >
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
              scope="row"
            >
              zep-spacing--xxl
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              48px
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              3rem
            </td>
            <td
              class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            >
              <div
                class="Spacing-square-0-1-9"
                style="width: 48px; height: 48px; color: green;"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="Spacing-table-0-1-1"
    >
      <caption
        class="visually-hidden"
      >
        Spacing Scale
      </caption>
      <thead
        class="Spacing-header-0-1-5"
      >
        <tr
          class="Spacing-tr-0-1-3"
        >
          <th
            class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-capitalize-0-1-7"
            scope="col"
          >
            name
          </th>
          <th
            class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-lowercase-0-1-6"
            scope="col"
          >
            px
          </th>
          <th
            class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-lowercase-0-1-6"
            scope="col"
          >
            rem
          </th>
          <th
            class="Spacing-th-0-1-2 zep-typo--normal-8 Spacing-capitalize-0-1-7"
            scope="col"
          >
            actual size
          </th>
        </tr>
      </thead>
      <tbody
        class="Spacing-tbody-0-1-4"
      >
        <tr
          class="Spacing-tr-0-1-3"
        >
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            scope="row"
          >
            zep-spacing--xxs
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            4px
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            0.25rem
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            <div
              class="Spacing-square-0-1-9"
              style="width: 4px; height: 4px; color: green;"
            />
          </td>
        </tr>
        <tr
          class="Spacing-tr-0-1-3"
        >
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            scope="row"
          >
            zep-spacing--xs
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            8px
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            0.5rem
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            <div
              class="Spacing-square-0-1-9"
              style="width: 8px; height: 8px; color: green;"
            />
          </td>
        </tr>
        <tr
          class="Spacing-tr-0-1-3"
        >
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            scope="row"
          >
            zep-spacing--s
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            12px
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            0.75rem
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            <div
              class="Spacing-square-0-1-9"
              style="width: 12px; height: 12px; color: green;"
            />
          </td>
        </tr>
        <tr
          class="Spacing-tr-0-1-3"
        >
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            scope="row"
          >
            zep-spacing--m
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            16px
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            1rem
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            <div
              class="Spacing-square-0-1-9"
              style="width: 16px; height: 16px; color: green;"
            />
          </td>
        </tr>
        <tr
          class="Spacing-tr-0-1-3"
        >
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            scope="row"
          >
            zep-spacing--l
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            24px
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            1.5rem
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            <div
              class="Spacing-square-0-1-9"
              style="width: 24px; height: 24px; color: green;"
            />
          </td>
        </tr>
        <tr
          class="Spacing-tr-0-1-3"
        >
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            scope="row"
          >
            zep-spacing--xl
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            32px
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            2rem
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            <div
              class="Spacing-square-0-1-9"
              style="width: 32px; height: 32px; color: green;"
            />
          </td>
        </tr>
        <tr
          class="Spacing-tr-0-1-3"
        >
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
            scope="row"
          >
            zep-spacing--xxl
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            48px
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            3rem
          </td>
          <td
            class="Spacing-th-0-1-2 Spacing-borderBottom-0-1-8 zep-typo--normal-7"
          >
            <div
              class="Spacing-square-0-1-9"
              style="width: 48px; height: 48px; color: green;"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
